src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff.Tool/Program.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Diagnostics;
+using Microsoft.DotNet.ApiSymbolExtensions.Logging;
+
+namespace Microsoft.DotNet.ApiDiff.Tool;
+
+/// <summary>
+/// Entrypoint for the genapidiff tool, which generates a markdown diff of two
+/// different versions of the same assembly, using the specified command line options.
+/// </summary>
+public static class Program
+{
+    public static async Task Main(string[] args)
+    {
+        RootCommand rootCommand = new("genapidiff");
+
+        Option<string> optionBeforeAssembliesFolderPath = new(["--before", "-b"])
+        {
+            Description = "The path to the folder containing the old (before) assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionBeforeRefAssembliesFolderPath = new(["--refbefore", "-rb"])
+        {
+            Description = "The path to the folder containing the old (before) reference assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionAfterAssembliesFolderPath = new(["--after", "-a"])
+        {
+            Description = "The path to the folder containing the new (after) assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionAfterRefAssembliesFolderPath = new(["--refafter", "-ra"])
+        {
+            Description = "The path to the folder containing the new (after) reference assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionOutputFolderPath = new(["--output", "-o"])
+        {
+            Description = "The path to the output folder.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionBeforeFriendlyName = new(["--beforeFriendlyName", "-bfn"])
+        {
+            Description = "The friendly name to describe the 'before' assembly.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionAfterFriendlyName = new(["--afterFriendlyName", "-afn"])
+        {
+            Description = "The friendly name to describe the 'after' assembly.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionTableOfContentsTitle = new(["--tableOfContentsTitle", "-tc"])
+        {
+            Description = "The title of the markdown table of contents file that is placed in the output folder.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
```

Comment:
Seems we could have a default for a lot of these things?  For instance if a single assembly were passed in we could probably come up with reasonable defaults.  Might be interesting to improve a lighter-weight "happy path"

Reply:
Sure, I think I could use a simple default and make this argument optional. The powershell script we use for the monthly api diffs will continue passing a value relevant to the diff comparison being done.

Reply:
Parameter is still marked as required.

Reply:
Fixed